Sources/SmileID/Classes/Networking/SmileIDService.swift

```diff
@@ -11,7 +10,7 @@ public protocol SmileIDServiceable {
     func prepUpload(request: PrepUploadRequest) async throws -> PrepUploadResponse
 
     /// Uploads files to S3. The URL should be the one returned by `prepUpload`.
-    func upload(zip: Data, to url: String) async throws -> UploadResponse
+    func upload(zip: Data, to url: String) async throws -> AsyncThrowingStream<UploadResponse, Error>
```

Comment:
How come we need to change the return type here?

Reply:
The return type on the main branch is `AnyPublisher<UploadResponse, Error>`. I removed the AnyPublisher wrapper in my previous PR but I later realised I needed to emit the `UploadSessionTask` progress from the delegate, that's why I changed it to `AsyncThrowStream<UploadResponse, Error>` which is more async/await friendly.
I'm trying as much as possible to preserve the functionalities we already have in the project even though I'm not sure we are actually displaying uploading progress in the UI.

Reply:
Got it, that makes sense. Yeah, we don't currently reflect the real time upload progress in the UI, but it could be something useful to add in the future